pkg/k8s/scanner/scanner.go

```diff
@@ -226,7 +226,7 @@ func (s *Scanner) scanK8sVulns(ctx context.Context, artifactsData []*artifacts.A
 	var resources []report.Resource
 	var nodeName string
 	if nodeName = findNodeName(artifactsData); nodeName == "" {
-		return nil, fmt.Errorf("failed to find node name")
+		return resources, nil
```

Comment:
Is it ok to return here? Can't we scan vulnerabilities at all when we don't find a node name?

Also, I found another place using `findNodeName`. Don't we need to fix it?
https://github.com/aquasecurity/trivy/blob/d9977c39ebf662815bd2426d80276a3b87013dd5/pkg/k8s/scanner/scanner.go#L355-L357

Reply:
1. yes, if there is no node name meaning its `namespace` or `resource` scope. k8s core components scanning is available only on cluster scope
2. the 2nd on is related to `kbom` if no node name then trivy show throw an error